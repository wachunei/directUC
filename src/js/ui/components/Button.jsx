import styled, { css } from "styled-components";

const Button = styled.button`
  border: 0;
  padding: 0.5rem 1rem;
  font-size: 12px;
  text-shadow: none;
  font-weight: 500;
  text-align: center;
  text-transform: uppercase;
  cursor: pointer;
  appearance: none;
  transition: filter 0.1s;

  ${(props) =>
    props.block &&
    css`
      display: block;
      width: 100%;
    `}

  background: ${(props) =>
    props.primary
      ? props.theme.buttonBackgroundPrimary
      : props.theme.buttonBackground};

  color: ${(props) =>
    props.primary
      ? props.theme.buttonForegroundPrimary
      : props.theme.buttonForeground};

  &:disabled {
    cursor: not-allowed;
    filter: grayscale(100%) opacity(80%);
  }

  &:focus {
    outline-width: 2px;
  }

  &&&:active {
    filter: brightness(95%);
  }

  &:hover:not(:disabled),
  &:focus:not(:disabled) {
    filter: brightness(105%);
  }
`;
export default Button;
