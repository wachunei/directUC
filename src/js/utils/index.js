export const removeSuffix = (suffix, value) =>
  value.endsWith(suffix) ? value.slice(0, -suffix.length) : value;

export const parseDistinguishedName = (name) =>
  name.replace(
    /\w\S*/g,
    (txt) => txt.charAt(0).toUpperCase() + txt.substr(1).toLowerCase()
  );

export const isFirefox = !window.chrome.app;
