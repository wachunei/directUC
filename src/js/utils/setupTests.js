// eslint-disable-next-line import/no-extraneous-dependencies
import "jest-styled-components";
