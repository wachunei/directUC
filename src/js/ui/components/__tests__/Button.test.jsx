import Button from "../Button";
import { render, getElement } from "./utils";

describe("Button Component", () => {
  test("default should match snapshot", () => {
    const { container } = render(<Button>Click Me!</Button>);
    expect(getElement(container)).toMatchSnapshot();
  });

  test("default should match snapshot with lightTheme", () => {
    const { container } = render(<Button>Click Me!</Button>, "lightTheme");
    expect(getElement(container)).toMatchSnapshot();
  });

  test("default should match snapshot with darkTheme", () => {
    const { container } = render(<Button>Click Me!</Button>, "darkTheme");
    expect(getElement(container)).toMatchSnapshot();
  });

  test("should render primary", () => {
    const { container } = render(<Button primary>Click Me!</Button>);
    expect(getElement(container)).toMatchSnapshot();
  });

  test("should render block", () => {
    const { container } = render(<Button block>Click Me!</Button>);
    expect(getElement(container)).toMatchSnapshot();
  });

  test("should render disabled", () => {
    const { container } = render(<Button disabled>Click Me!</Button>);
    expect(getElement(container)).toMatchSnapshot();
  });

  test("should render primary block", () => {
    const { container } = render(
      <Button primary block>
        Click Me!
      </Button>
    );
    expect(getElement(container)).toMatchSnapshot();
  });
});
