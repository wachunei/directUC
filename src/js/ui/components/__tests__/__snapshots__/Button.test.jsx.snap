// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component default should match snapshot 1`] = `
.c0 {
  border: 0;
  padding: 0.5rem 1rem;
  font-size: 12px;
  text-shadow: none;
  font-weight: 500;
  text-align: center;
  text-transform: uppercase;
  cursor: pointer;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  -webkit-transition: filter 0.1s;
  transition: filter 0.1s;
  background: #E0E0E0;
  color: #505050;
}

.c0:disabled {
  cursor: not-allowed;
  -webkit-filter: grayscale(100%) opacity(80%);
  filter: grayscale(100%) opacity(80%);
}

.c0:focus {
  outline-width: 2px;
}

.c0.c0:not(:disabled):active {
  -webkit-filter: brightness(95%);
  filter: brightness(95%);
}

.c0:hover:not(:disabled),
.c0:focus:not(:disabled) {
  -webkit-filter: brightness(105%);
  filter: brightness(105%);
}

<button
  className="c0"
>
  Click Me!
</button>
`;

exports[`Button Component default should match snapshot with darkTheme 1`] = `
.c0 {
  border: 0;
  padding: 0.5rem 1rem;
  font-size: 12px;
  text-shadow: none;
  font-weight: 500;
  text-align: center;
  text-transform: uppercase;
  cursor: pointer;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  -webkit-transition: filter 0.1s;
  transition: filter 0.1s;
  background: #505050;
  color: #E0E0E0;
}

.c0:disabled {
  cursor: not-allowed;
  -webkit-filter: grayscale(100%) opacity(80%);
  filter: grayscale(100%) opacity(80%);
}

.c0:focus {
  outline-width: 2px;
}

.c0.c0:not(:disabled):active {
  -webkit-filter: brightness(95%);
  filter: brightness(95%);
}

.c0:hover:not(:disabled),
.c0:focus:not(:disabled) {
  -webkit-filter: brightness(105%);
  filter: brightness(105%);
}

<button
  className="c0"
>
  Click Me!
</button>
`;

exports[`Button Component default should match snapshot with lightTheme 1`] = `
.c0 {
  border: 0;
  padding: 0.5rem 1rem;
  font-size: 12px;
  text-shadow: none;
  font-weight: 500;
  text-align: center;
  text-transform: uppercase;
  cursor: pointer;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  -webkit-transition: filter 0.1s;
  transition: filter 0.1s;
  background: #E0E0E0;
  color: #505050;
}

.c0:disabled {
  cursor: not-allowed;
  -webkit-filter: grayscale(100%) opacity(80%);
  filter: grayscale(100%) opacity(80%);
}

.c0:focus {
  outline-width: 2px;
}

.c0.c0:not(:disabled):active {
  -webkit-filter: brightness(95%);
  filter: brightness(95%);
}

.c0:hover:not(:disabled),
.c0:focus:not(:disabled) {
  -webkit-filter: brightness(105%);
  filter: brightness(105%);
}

<button
  className="c0"
>
  Click Me!
</button>
`;

exports[`Button Component should render block 1`] = `
.c0 {
  border: 0;
  padding: 0.5rem 1rem;
  font-size: 12px;
  text-shadow: none;
  font-weight: 500;
  text-align: center;
  text-transform: uppercase;
  cursor: pointer;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  -webkit-transition: filter 0.1s;
  transition: filter 0.1s;
  display: block;
  width: 100%;
  background: #E0E0E0;
  color: #505050;
}

.c0:disabled {
  cursor: not-allowed;
  -webkit-filter: grayscale(100%) opacity(80%);
  filter: grayscale(100%) opacity(80%);
}

.c0:focus {
  outline-width: 2px;
}

.c0.c0:not(:disabled):active {
  -webkit-filter: brightness(95%);
  filter: brightness(95%);
}

.c0:hover:not(:disabled),
.c0:focus:not(:disabled) {
  -webkit-filter: brightness(105%);
  filter: brightness(105%);
}

<button
  className="c0"
>
  Click Me!
</button>
`;

exports[`Button Component should render disabled 1`] = `
.c0 {
  border: 0;
  padding: 0.5rem 1rem;
  font-size: 12px;
  text-shadow: none;
  font-weight: 500;
  text-align: center;
  text-transform: uppercase;
  cursor: pointer;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  -webkit-transition: filter 0.1s;
  transition: filter 0.1s;
  background: #E0E0E0;
  color: #505050;
}

.c0:disabled {
  cursor: not-allowed;
  -webkit-filter: grayscale(100%) opacity(80%);
  filter: grayscale(100%) opacity(80%);
}

.c0:focus {
  outline-width: 2px;
}

.c0.c0:not(:disabled):active {
  -webkit-filter: brightness(95%);
  filter: brightness(95%);
}

.c0:hover:not(:disabled),
.c0:focus:not(:disabled) {
  -webkit-filter: brightness(105%);
  filter: brightness(105%);
}

<button
  className="c0"
  disabled={true}
>
  Click Me!
</button>
`;

exports[`Button Component should render primary 1`] = `
.c0 {
  border: 0;
  padding: 0.5rem 1rem;
  font-size: 12px;
  text-shadow: none;
  font-weight: 500;
  text-align: center;
  text-transform: uppercase;
  cursor: pointer;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  -webkit-transition: filter 0.1s;
  transition: filter 0.1s;
  background: #1F95EB;
  color: #FFFFFF;
}

.c0:disabled {
  cursor: not-allowed;
  -webkit-filter: grayscale(100%) opacity(80%);
  filter: grayscale(100%) opacity(80%);
}

.c0:focus {
  outline-width: 2px;
}

.c0.c0:not(:disabled):active {
  -webkit-filter: brightness(95%);
  filter: brightness(95%);
}

.c0:hover:not(:disabled),
.c0:focus:not(:disabled) {
  -webkit-filter: brightness(105%);
  filter: brightness(105%);
}

<button
  className="c0"
>
  Click Me!
</button>
`;

exports[`Button Component should render primary block 1`] = `
.c0 {
  border: 0;
  padding: 0.5rem 1rem;
  font-size: 12px;
  text-shadow: none;
  font-weight: 500;
  text-align: center;
  text-transform: uppercase;
  cursor: pointer;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  -webkit-transition: filter 0.1s;
  transition: filter 0.1s;
  display: block;
  width: 100%;
  background: #1F95EB;
  color: #FFFFFF;
}

.c0:disabled {
  cursor: not-allowed;
  -webkit-filter: grayscale(100%) opacity(80%);
  filter: grayscale(100%) opacity(80%);
}

.c0:focus {
  outline-width: 2px;
}

.c0.c0:not(:disabled):active {
  -webkit-filter: brightness(95%);
  filter: brightness(95%);
}

.c0:hover:not(:disabled),
.c0:focus:not(:disabled) {
  -webkit-filter: brightness(105%);
  filter: brightness(105%);
}

<button
  className="c0"
>
  Click Me!
</button>
`;
